pkg/controller/scyllacluster/resource.go

```diff
@@ -110,6 +110,33 @@ func MemberService(sc *scyllav1.ScyllaCluster, rackName, name string, oldService
 		}
 	}
 
+	var internalTrafficPolicy *corev1.ServiceInternalTrafficPolicy
+	var externalTrafficPolicy corev1.ServiceExternalTrafficPolicy
+	var allocateLoadBalancerNodePorts *bool
+	var loadBalancerClass *string
+	serviceType := corev1.ServiceTypeClusterIP
+	clusterIP := ""
+
+	if sc.Spec.ExposeOptions != nil && sc.Spec.ExposeOptions.NodeService != nil {
+		ns := sc.Spec.ExposeOptions.NodeService
+
+		switch ns.Type {
+		case scyllav1.NodeServiceTypeClusterIP:
+			serviceType = corev1.ServiceTypeClusterIP
+		case scyllav1.NodeServiceTypeLoadBalancer:
+			serviceType = corev1.ServiceTypeLoadBalancer
+		case scyllav1.NodeServiceTypeHeadless:
+			serviceType = corev1.ServiceTypeClusterIP
+			clusterIP = corev1.ClusterIPNone
```

Comment:
nit: move to a separate block for readability
```go
		switch ns.Type {
		case scyllav1.NodeServiceTypeClusterIP:
			serviceType = corev1.ServiceTypeClusterIP
		case scyllav1.NodeServiceTypeLoadBalancer:
			serviceType = corev1.ServiceTypeLoadBalancer
		case scyllav1.NodeServiceTypeHeadless:
			serviceType = corev1.ServiceTypeClusterIP
                }

                if ns.Type == scyllav1.NodeServiceTypeHeadless {
			clusterIP = corev1.ClusterIPNone
                }
```

Reply:
good idea, done

Reply:
interesting :) I'd have expected to set all options specific to this service type at the `case` with the service type

Reply:
I think my preference comes from being used to pattern matching, where you'd just assign the result of the case like:
```
let serviceType = case ns.Type of
        scyllav1.NodeServiceTypeClusterIP -> corev1.ServiceTypeClusterIP
        ...
```
Personally I think adding an extra assignment in just one of the cases is easy to miss, but I do agree that it's somewhat redundant to add an extra check later. Idk, the closest thing to the above in Go would probably be a function converting NodeServiceType to ServiceType?

Reply:
brought back initial code, if list of stuff needed to be done for particular case, having them all in one branch is more readable.